src/sync.d

```diff
@@ -1744,11 +1747,41 @@ final class SyncEngine
 		try {
 			onedrive.deleteById(item.driveId, item.id, item.eTag);
 		} catch (OneDriveException e) {
-			if (e.httpStatusCode == 404) log.vlog("OneDrive reported: The resource could not be found.");
-			else throw e;
+			if (e.httpStatusCode == 404) {
+				// item.id, item.eTag could not be found on driveId
+				log.vlog("OneDrive reported: The resource could not be found.");
+			}
+			
+			else {
+				// Not a 404 response .. is this a 403 response due to OneDrive Business Retention Policy being enabled?
+				if ((e.httpStatusCode == 403) && (accountType != "personal")) {
+					auto errorArray = splitLines(e.msg);
+					JSONValue errorMessage = parseJSON(replace(e.msg, errorArray[0], ""));
+					if (errorMessage["error"]["message"].str == "Request was cancelled by event received. If attempting to delete a non-empty folder, it's possible that it's on hold") {
+						// Issue #338 - Unable to delete OneDrive content when OneDrive Business Retention Policy is enabled
+						// TODO: We have to recursively delete all files & folders from this path to delete
+						// WARN: 
+						log.error("\nERROR: Unable to delete the requested remote path from OneDrive: ", path);
+						log.error("ERROR: This error is due to OneDrive Business Retention Policy being applied");
+						log.error("WORKAROUND: Manually delete all files and folders from the above path as per Business Retention Policy\n");
+					}
```

Comment:
Isn't here a `return` missing? If that error happens, we will continue out of the `catch` block and go into the following code, or am I missing something?

Reply:
Before there was a `throw` again, but now we seem to continue after the catch block

Reply:
Essentially what we are doing here is printing out details / information rather than crashing the application which is what the `throw e` would do.

For the 403 & business handling - this is the correct response - print message and continue (resolves user issue). If the error is not a 404 or a 403 + business, rather than crash, report what the error was and return out of the loop.

IMHO this is better than crashing the application ..

Reply:
Ok, just wanted to check back